src/drivers/Hrs3300.cpp

```diff
@@ -67,40 +67,27 @@ void Hrs3300::Disable() {
   WriteRegister(static_cast<uint8_t>(Registers::PDriver), 0);
 }
 
-uint32_t Hrs3300::ReadHrs() {
-  auto m = ReadRegister(static_cast<uint8_t>(Registers::C0DataM));
-  auto h = ReadRegister(static_cast<uint8_t>(Registers::C0DataH));
-  auto l = ReadRegister(static_cast<uint8_t>(Registers::C0dataL));
-  return ((l & 0x30) << 12) | (m << 8) | ((h & 0x0f) << 4) | (l & 0x0f);
-}
-
-uint32_t Hrs3300::ReadAls() {
-  auto m = ReadRegister(static_cast<uint8_t>(Registers::C1dataM));
-  auto h = ReadRegister(static_cast<uint8_t>(Registers::C1dataH));
-  auto l = ReadRegister(static_cast<uint8_t>(Registers::C1dataL));
-  return ((h & 0x3f) << 11) | (m << 3) | (l & 0x07);
-}
-
-void Hrs3300::SetGain(uint8_t gain) {
-  constexpr uint8_t maxGain = 64U;
-  gain = std::min(gain, maxGain);
-  uint8_t hgain = 0;
-  while ((1 << hgain) < gain) {
-    ++hgain;
+Hrs3300::PackedHrsAls Hrs3300::ReadHrsAls() {
+  uint8_t buf[8];
+  uint8_t base = static_cast<uint8_t>(Registers::C1dataM);
```

Comment:
I mean if you really wanted it would be possible to compute the relative offsets at compile time with constexpr. 
```c++
  constexpr Registers dataRegisters[] =
    {Registers::C1dataM, Registers::C0DataM, Registers::C0DataH, Registers::C1dataH, Registers::C1dataL, Registers::C0dataL};
  constexpr uint8_t base = static_cast<uint8_t>(*std::min_element(std::begin(dataRegisters), std::end(dataRegisters)));
  ```
Would work.

But it seems contrived to me as the register locations are never going to change for this chip, and this file is a chip specific driver

Reply:
yes, the constexpr solution seems very nice. Otherwise `base` just seemed magic to me when reading the changes for the first few times. With constexpr the way `base` is determined is now explicit. And we don't have runtime overhead for this computation. Win-Win in my opinion :)

As I'm not much in the drivers area I may need more "handholding" than others :sweat_smile: